src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java

```diff
@@ -44,20 +44,41 @@ void extract(Table table, BigQueryExtractionData extractionData) {
   }
 
   void cleanup(BigQueryExtractionData extractionData) {
-    String dataBucket = extractionData.getDataBucket();
-    String dataKey = extractionData.getDataKey();
-    String format = extractionData.getFormat();
-    String dataUri = extractionData.getDataUri();
+    deleteBucketAndContents(extractionData.getDataBucket());
+  }
+
+  private void deleteBucketAndContents(String dataBucket) {
+    deleteObjectsInBucket(dataBucket);
+    deleteBucket(dataBucket);
+  }
 
-    BlobId blobId = BlobId.of(dataBucket, dataKey + "." + format);
-    boolean suceeded = storage.delete(blobId);
-    if (!suceeded) {
-      log.warn("Could not delete object {}", dataUri);
+  private void deleteObjectsInBucket(String dataBucket) {
+    try {
+      Iterable<Blob> blobs = storage.list(dataBucket).iterateAll();
+      for (Blob blob : blobs) {
+        boolean suceeded = storage.delete(blob.getBlobId());
+        if (suceeded) {
+          log.info("Deleted object {}", blob);
+        } else {
+          log.warn("Could not delete object {}", blob);
+        }
+      }
+    } catch (Exception e) {
```

Comment:
can't we catch anything more specific? 
Should we move the try-catch inside the for loop so around every delete, so all deletes are attempted?

Reply:
Above would be nice, @courtsVII when you have a chance and if the above look viable I'd suggest either a small PR with just this change, or alternatively a ticket describing this so we can do it later.